### PR TITLE
Revise humidity component

### PR DESCRIPTION
**Pull request by lkluft**
This PR:
* **Breaks** old control scripts as it introduces a new interface to the humidity component
* Separates the models that describe the vertical relative humidity distribution on the hand and models that describe the stratospheric coupling on the other hand

I am not completely sure if that is the final form (especially when it comes to naming of the classes, should we introduce suffixes e.g. ``Manabe67_RH``?). Anyhow, I want to know your opinion on the general approach. My main goals were to _1)_ make the single classes more readable and _2)_ allow for a more flexible configuration of the humidity model.

### CODE REVIEW

File: konrad/core.py [hunk 1]
```diff
@@ -89,8 +88,8 @@ def __init__(self, atmosphere, timestep='3h', max_duration='5000d',
         self.ozone = utils.return_if_type(ozone, 'ozone',
                                           Ozone, OzonePressure())
 
-        self.humidity = utils.return_if_type(humidity, 'humidity',
-                                             Humidity, FixedRH())
+        if humidity is None:
+            self.humidity = FixedRH()
```
Comment (SallyDa) [inline konrad/core.py]: You need
`else:
     self.humidity = humidity`
after this, in case we want to use something other than the default.

Comment (lkluft) [reply konrad/core.py]: Yes, of course!

**Comment by lkluft** [comment]
Thanks! :relaxed: Are you fine with the class names? I am not sure if we should try to make them more distinguishable because all are visible in the ``konrad.humidity`` namespace.

_PS.: I want to rebase before the merging 😉 So please don't merge yourself 😄_

**Comment by SallyDa** [comment]
Yeah, I'm fine with the class names. I see what you mean about the namespace but it's fine for now, you can always change it later if you come up with a good solution for that.

**Pull request merged**